src/engine/QueryPlanner.cpp

```diff
@@ -767,8 +767,21 @@ auto QueryPlanner::seedWithScansAndText(
       SubtreePlan newIdPlan = plan;
       // give the plan a unique id bit
       newIdPlan._idsOfIncludedNodes = uint64_t(1) << idShift;
-      newIdPlan._idsOfIncludedFilters = 0;
-      newIdPlan.idsOfIncludedTextLimits_ = 0;
+
+      // Either the _idsOfIncludedFilters and idsOfIncludedTextLimits_ of the
+      // plan are all `0`, or the plan is either a MINUS, OPTIONAL, or BIND (for
+      // which we have special handling).
+      AD_CORRECTNESS_CHECK(
+          (newIdPlan._idsOfIncludedFilters == 0 &&
+           newIdPlan.idsOfIncludedTextLimits_ == 0) ||
+              std::dynamic_pointer_cast<Bind>(
+                  newIdPlan._qet->getRootOperation()) ||
+              std::dynamic_pointer_cast<OptionalJoin>(
+                  newIdPlan._qet->getRootOperation()) ||
+              std::dynamic_pointer_cast<Minus>(
+                  newIdPlan._qet->getRootOperation()),
```

Comment:
Add a lambda for this, like

`auto is = [](auto ti) { 
using T = typename decltype(ti)::type;
return dynamic_cast<const T*>(newIdPlan._qect->getRootOperation().get()) != nullptr;
};
using namespace ad_utility::use_type_identity;  (see the corresponding header for example usages)
then
`bool isBind = is(ti<Bind>)`  (similar for the other two in the large condition).